mcs/class/referencesource/mscorlib/system/rttype.cs

```diff
@@ -4258,6 +4258,10 @@ public override Type MakeGenericType(Type[] instantiation)
 
                 if (rtInstantiationElem == null)
                 {
+#if MONO                    
+                    if (instantiationElem.IsSignatureType)
+                        return Internal.Reflection.Augments.ReflectionAugments.MakeGenericSignatureType (this, instantiation);
```

Comment:
Shouldn't mono also return a `TypeBuilderInstantiation`?

Reply:
I've just found that there is the difference between dotnet/corefx and mono/corefx.
`Type` in upstream contains [MakeGenericSignatureType ](https://github.com/dotnet/corefx/blob/master/src/Common/src/CoreLib/System/Type.cs#L349)method and Mono fork doesn't. So I need to update `Type` in mono/corefx.